src/apdu_sign.c

```diff
@@ -354,6 +356,14 @@ void parse_operation(struct maybe_transaction* _U_ dest, bip32_path_t* _U_ key_d
 	if(mol_result != MOL_OK)
 	  REJECT("Transaction verification returned %d; parse failed\nbody: %.*h\n", mol_result, buff_size, buff);
 
+	parse_operation_inner(dest, key_derivation, buff, buff_size);
+}
+
+void parse_operation_inner(struct maybe_transaction* _U_ dest, bip32_path_t* _U_ key_derivation, uint8_t *const buff, uint16_t const buff_size) {
```

Comment:
It looks like this is to avoid MolReader_RawTransaction_verify allocating too much to the stack? I'm not too familiar with how this works in C, but I wonder if wrapping this in a {} block would also address this:

```
{
uint8_t mol_result=MolReader_RawTransaction_verify(&seg,true);
if(mol_result != MOL_OK)
  REJECT("Transaction verification returned %d; parse failed\nbody: %.*h\n", mol_result, buff_size, buff);
}
```

Reply:
Although since it works as is, it sounds fair to keep it.

Reply:
Wrapping it (or rather, wrapping the rest of the function) in a { } did not produce that effect so far as I could see, which is why I took this approach. It just didn't fit after the interpretation section's local variables, and local variables were being allocated for the whole function right at the start, including the scoped block. I would not be surprised to discover that this part of stack behavior was not specified strongly in the base C spec.